daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/InfosetImpl.scala

```diff
@@ -1415,6 +1423,50 @@ sealed class DIComplex(override val erd: ElementRuntimeData, val tunable: Daffod
     }
   }
 
+  final def sortChildNodesByPosition(): Unit = {
+    // TODO: Once we upgrade to Scala 2.13 we can call the inplace sort method: DAFFODIL-2152
+    // childNodes = childNodes.sortInPlace(_.erd.position)
```

Comment:
I think this is fine, as it only gets called by the unordered parser, which means it will only sort the children in the unordered sequence.  In the tests I have, including tests where the unordered sequence is part of a larger sequence, childNodes only contains the elements of the unordered sequence.

Reply:
Are you sure? If you have something like:
```xml
<xs:element name="root">
  <xs:complexType>
    <xs:sequence>
      <xs:element name="a" />
      <xs:element name="b" />
      <xs:sequence dfdl:sequenceKind="unordered">
        <xs:element name="c" />
        <xs:element name="d" />
     </xs:sequence>
  </xs:complexType>
</xs:element>
```
In that case elements a, b, c, and d are all children of the root element, and I would think would all be in the same childNodes array.